pennylane/ops/op_math/controlled_class.py

```diff
@@ -166,6 +166,9 @@ def __new__(cls, base, *_, **__):
     def __init__(
         self, base, control_wires, control_values=None, work_wires=None, do_queue=True, id=None
     ):
+        if control_wires is None:
+            raise ValueError("Must specify control wires")
+
```

Comment:
Why this check? Did you find a corner case where `control_wires=None`?

Reply:
> 1. Please try to separate "different" tasks in different PRs. Especially when moving code around: if you create a PR where you ONLY move the code you will save the reviewer a lot of time.
> 2. I assumed that the `ControlledQubitUnitary` class and tests were ONLY moved, thus I didn't check its code.

Sorry, this did get a bit messy. I guess I should clarify that the primary purpose of this PR was for ControlledQubitUnitary to inherit as much behaviour as possible from Controlled (instead of having it defined in ControlledQubitUnitary), so the primary changes being made here are changing Controlled and ControlledQubitUnitary. So its definitely **not** the case that the ControlledQubitUnitary class was only moved. 

Moving the code between files was an unfortunate necessity to avoid circular imports between `math_ops` and `qubit folders`. I'll aim to make things like that a separate initial PR in the future.

Adding parameter broadcasting is also a subtask that was needed to make sure that ControlledQubitUnitary didn't lose any features when being modified to inherit its behaviour from Controlled. Here are the changes specific to adding broadcasting: [#3577](https://github.com/PennyLaneAI/pennylane/pull/3577). This probably also should have been a separate PR if possible.

Reply:
```pycon
>>> qml.ops.Controlled(qml.PauliX(0))
TypeError: ControlledOp.__init__() missing 1 required positional argument: 'control_wires'
```
`ControlledQubitUnitary` had that check because the call signature had a default value for `control_wires`. We no longer have a default value, because it is an argument instead of a keyword argument.

I'm Ok with just turning a `ValueError` into a `TypeError` and getting rid of this logic.